src/TLuaInterpreter.cpp

```diff
@@ -10320,6 +10347,19 @@ int TLuaInterpreter::setDefaultAreaVisible(lua_State* L)
     return 1;
 }
 
+int TLuaInterpreter::getGuiLanguageCode(lua_State* L)
+{
+    const QString languageCode(mudlet::self()->getGuiLanguage());
+
+    if (languageCode.isEmpty()) {
+        lua_pushstring(L, "none");
```

Comment:
What about `nil` instead?

Reply:
💭 Well that would be _one_ way to do it - but then a script would have to respond to the _nil_ specially - OTOH it might be worth it so that _nil_ means that there ARE no translations available as opposed to "none" meaning it has been "switched-off"...

Reply:
I don't see how that would matter to an API. Let's go with `nil`, if we get requests specifically for "none" we can add it.

Reply:
I have only give a getter for the language (because it is common across all active profiles and it is not something that a profile should be able to change, only to detect) but if there were to be a setter as well it might be worth returning a `nil` when there are __no__ translations available but an English as per Lua API requirements "none" when there are but they have been disabled... does that seem sensible even thought I am not planning a setter for GUI Language?

Reply:
Actually neither `nil` nor `none` make sense. You can't have "no language". It'll be English at minimum...

Reply:
Well, it depends on whether you always want an en_US translation (which is plurals only) or the source code text (if you are debugging a translation) - the difference is when the `%n' replacement parameter is used in `tr(`...`)` see the "Handling Plurals" section in [I18n source translation](http://doc.qt.io/qt-5/i18n-source-translation.html)

In both the "no translations available" or the "none selected case" uses will see messages like:

    0 Room(s) selected
    1 Room(s) selected
    2 Room(s) selected

but if the en_US translation is available this should change to the cleaner/clearer:

    0 Rooms selected
    1 Room selected
    2 Rooms selected

whilst the end user may never want to run with out a translation loaded, we developers likely _will_ want the ability to turn off the translation infrastructure...!  From a diagnostic point of view a package creator who is contemplating supporting multiple languages they may want to know the difference between knowing that the user does have but is not using a particular translation compared to a possibly mis-configured setup where the executable cannot find the auxiallary translation files.

:bulb: As well as providing the current translation code (and the previous one in the Mudlet change event) perhaps we should also provide a list of the codes available...?

Reply:
As we have always had, there always has been a language available - so `en_US` will always be at least available. Thus there should never be `nil` or `none` returned I think is the way to go.

If a developer wants to debug this, they can fiddle around with it as needed. If we find we need to do it often, we'll put in tools - but right now we have no idea and let's err on the side of less work, less maintenance, less issues.

Reply:
> As we have always had, there always has been a language available

I am trying to say that is not exactly the case - we have always had the texts included in the source code and that is English-ish and largely to the American version - however we will be including a translation explicitly for that locale which only differs in the handling of plurals (to make them look nicer) but the latter is only going to be available provided the translations are found when __mudlet__ starts up, should something fail in that process (partial install, permissions problems, etc.) the executable will fall-back to running without, (admittedly that is not likely to be an issue for those users/platforms that accept our likely action to include the translations in the resources)...

Reply:
Yeah but even in that case, Mudlet is will be using some language, not `none` or `nil` - so we can assume it to be `en_US`, even if it's not technically 100% correct. Nobody will mind, and it'll reduce the confusion we'd get from seeing a `none` language

Reply:
This still needs to change - "none" is not a language people speak! Let's make this for humans, not engineers :wink: